girder_worker/utils.py

```diff
@@ -369,34 +369,81 @@ def load_plugin(name, paths):
                 name, '\n   '.join(paths)))
 
 
-def run_process(command, outputs, print_stdout, print_stderr):
+def run_process(command, outputs, print_stdout, print_stderr,
+                output_pipes=None):
```

Comment:
@jeffbaumes this function is now too complex for our style checker, but I was thinking of breaking it anyway. I think the new signature should just be:

    run_process(command, output_pipes)

This unifies the behavior regardless of whether the stream is stdout, stderr, or a named pipe. It would be up to the caller to pass in adapters that would handle each stream. I think that's a cleaner interface and certainly a cleaner implementation, though it would technically be breaking. What do you think?